Calculator/ViewController.swift

```diff
@@ -45,16 +47,26 @@ class ViewController: UIViewController {
             self?.updateDisplay(self?.calculator.displayText)
             }).addDisposableTo(self.disposableBag)
 
-        self.squareRootButton.rx_tap.subscribeNext ({ [weak self] _ in
-
+        self.squareRootButton.rx_tap.subscribeNext ({ [weak self] _ in 
             self?.calculator.calculate(.SquareRoot)
             self?.updateDisplay(self?.calculator.displayText)
         }).addDisposableTo(self.disposableBag)
 
-        self.dotButton.rx_tap.subscribeNext({ [weak self] in
-            self?.updateDisplay(self?.calculator.addDotToNumber((self?.displayLabel.text!)!))
-
+        self.dotButton.rx_tap.subscribeNext({ [weak self] _ in
+            if let text = self?.displayLabel.text {
+                self?.updateDisplay(self?.calculator.addDotToNumber(text))
+            }
             }).addDisposableTo(self.disposableBag)
+
+        self.clearButton.rx_tap.subscribeNext ({ [weak self] _ in
+            self?.calculator.reset()
+            self?.updateDisplay(self?.calculator.displayText)
+        }).addDisposableTo(self.disposableBag)
+
+        self.calculator.optStackSubject.asObservable().subscribeNext ({ (array) in
```

Comment:
I would try to hide the implementation more here, you're exposing your privates a bit :) perhaps the calculator can expose a `func optStact() -> Observable<[Op]>`